manifests/users.pp

```diff
@@ -40,22 +41,31 @@
     value   => $ca_server,
   }
 
-  if $metrics_server {
-    file { "/home/${user}/time-puppet-run.sh":
-      ensure => file,
-      content => "TIMEFORMAT=\"metrics.${::fqdn}.${user}.time %R `date +%s`\"; TIME=$( { time /opt/puppet/bin/puppet agent --onetime --no-daemonize > /dev/null; } 2>&1 ); echo \$TIME | nc ${metrics_server} ${metrics_port}",
+  if $daemonize {
+    # TODO: we need to capture run timings for publication to graphite
+
+    exec { "user ${user} daemon puppet agent":
+      command => "/usr/bin/sudo -u ${user} /opt/puppet/bin/puppet agent --daemonize >/dev/null 2>&1",
```

Comment:
I'm not thrilled here.  The intent is "have the non-root user(s) run this version of the agent". It works using `sudo`.

There are two significant downsides:

 1. Um, `sudo`.
 2. `sudo` on CentOS will fail when there is no controlling terminal, unless one modifies `/etc/sudoers` ([as documented here](https://www.shell-tips.com/2014/09/08/sudo-sorry-you-must-have-a-tty-to-run-sudo/))

/cc @doug-rosser @ericwilliamson